src/syspurpose/files.py

```diff
@@ -58,12 +58,16 @@ def create(self):
 
     def add(self, key, value):
         """
-        Add a value to a list of values specified by key
+        Add a value to a list of values specified by key. If the current value specified by the key is scalar/non-list,
+        it is not overridden, but maintained in the list, along with the new value.
         :param key: The name of the list
         :param value: The value to append to the list
         :return: None
         """
         try:
+            current_value = self.contents[key]
+            if current_value is not None and not isinstance(current_value, list):
+                self.contents[key] = [current_value]
             self.contents[key].append(value)
```

Comment:
If we want unique values for properties that are lists, we could also use the `set()` data structure and simply call `.add` on it. The question is though, do we want to eliminate duplicate values, or do we want to allow them?

Reply:
We don't have any planned fields that need duplicate values, so let's use set(), and then when producing the JSON, it would be nice to use `sorted()`, so we can get predictable ordering.

Reply:
I agree with kahowell's suggestion.

Reply:
I'm trying to use `set()`, but its getting ugly since sets aren't serializable by default, custom serializer functions are required every time serialization/deserialization happens to transform sets from/to lists. I'm thinking keeping the lists but adding the check in is much simpler and cleaner.
@cnsnyder @kahowell